### ISSUE TITLE
Crash on Electron apps

### ISSUE THREAD

**Issue by crimist** [FancyWM/fancywm-issues#27]
FancyWM seems to crash for me when performing various actions on Electron based programs.

Examples:
* Discord
* Spotify

## Reproduction

Unfloating Electron app window (100% reproducible)

* Float window `Shift+Super+F`
* Unfloat (restore) window `Shift+Super+F` (crashes)

Move Electron app window (crashes occasionally):

* Merge window into avaiable split zone

Possibly more cases but I can't use the software long enough to find any others as it keeps crashing.
 
## SysInfo

Windows 10 Enterprise x64 21H1 (build 19043.1466)

## Stacktraces

```
System.ArgumentException: An item with the same key has already been added. Key: {000301C8: Chrome_WidgetWin_1}
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at FancyWM.TilingService.<OnWindowStateChanged>g__UnregisterAndSaveLocation|107_0(<>c__DisplayClass107_0& )
   at FancyWM.TilingService.OnWindowStateChanged(Object sender, WindowStateChangedEventArgs e)
   at WinMan.Windows.Win32Window.<>c__DisplayClass129_0.<UpdateStateAndNotify>b__0()
   at WinMan.Windows.Win32Window.UpdateConfiguration()
   at WinMan.Windows.Win32Window.OnPositionChanged()
   at WinMan.Windows.Utilities.EventLoop.Run()
--- End of stack trace from previous location ---
   at FancyWM.Utilities.Dispatchers.<>c__DisplayClass4_0.<RethrowOnDispatcher>b__0()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run()
   at FancyWM.Startup.Main(String[] args)
```

```
System.ArgumentException: An item with the same key has already been added. Key: {0001078A: Chrome_WidgetWin_0}
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at FancyWM.TilingService.<OnWindowStateChanged>g__UnregisterAndSaveLocation|107_0(<>c__DisplayClass107_0& )
   at FancyWM.TilingService.OnWindowStateChanged(Object sender, WindowStateChangedEventArgs e)
   at WinMan.Windows.Win32Window.<>c__DisplayClass129_0.<UpdateStateAndNotify>b__0()
   at WinMan.Windows.Win32Window.UpdateConfiguration()
   at WinMan.Windows.Win32Window.OnPositionChanged()
   at WinMan.Windows.Utilities.EventLoop.Run()
--- End of stack trace from previous location ---
   at FancyWM.Utilities.Dispatchers.<>c__DisplayClass4_0.<RethrowOnDispatcher>b__0()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run()
   at FancyWM.Startup.Main(String[] args)
```

**Comment by veselink1** [comment]
Thanks for reporting this! Should be fixed in v2.4.1 now in the Microsoft Store.

**Comment by crimist** [comment]
Confirmed fixed, thanks.
